### DISCUSSION TITLE
Copilot emacs eglot integration initial implementation

### DISCUSSION THREAD

**Discussion by tyler-dodge** [community/community#33849] [Copilot Conversations]
Hi, I made an initial implementation of a copilot plugin for emacs using eglot for LSP. It currently supports:
* Authentication
* Generating completion results based on editor state
* Refreshing a completion panel
* Sorting all company backends based on whether or not the completion panel contains a token from the company result.

I'd like to contribute it back to you all if you're interested. I'm not sure the proper channel for contacting the dev team besides this. I haven't applied a license yet to it (it can be whatever you all prefer.) 

I was also planning on doing a little more productionization prior to release to any of the package repos. 

Initial Implementation: https://github.com/tyler-dodge/eglot-copilot/blob/main/copilot.el

**Comment by iamcjwilliams** [comment]
@tyler-dodge Hello! Thank you for your feedback! Our copilot team will check it out!

**Comment by milanglacier** [comment]
Currently eglot doesn't support multiple lsp servers attached to same buffer. Implementing it with eglot might not be a good idea. Usually people need a real LSP (like jdtls) working with copilot together.

**Reply by tyler-dodge** [reply]
Yeah that’s why it uses a shadow buffer for copilot, so that the actual buffer can be using a different LSP server while still receiving the suggestions from copilot.
